rust/core-lib/src/plugins/manager.rs

```diff
@@ -145,6 +146,41 @@ impl PluginManager {
         }
     }
 
+    fn request_plugins_sync(&self, view_id: ViewIdentifier,
```

Comment:
Looking down at the code where you create a dummy view identifier, I don't think we should add this method. It's only used (and probably should only ever be used) for collecting traces, but by presenting itself as a general helper method it includes fields (`view_id` and `only_globals`) that are irrelevant to this use case. Instead the plugin manager should have a method specific to this request, `collect_traces`, which implements this specific logic.

Also given that it's a slow, blocking call, we really don't want anyone else to reuse it.

Reply:
Makes sense.  I'll adjust & put a note that eventually should migrate to an asynchronous API.